libs/map-editor/src/types.ts

```diff
@@ -1,9 +1,14 @@
 import { ITiledMapObject } from "@workadventure/tiled-map-type-guard";
+import { CreateAreaCommandConfig } from './Commands/Area/CreateAreaCommand';
+import { DeleteAreaCommandConfig } from './Commands/Area/DeleteAreaCommand';
 import { UpdateAreaCommandConfig } from './Commands/Area/UpdateAreaCommand';
 
 export type ITiledMapRectangleObject = ITiledMapObject & { width: number; height: number };
 
-export type CommandConfig = UpdateAreaCommandConfig;
+export type CommandConfig =
+    UpdateAreaCommandConfig |
+    DeleteAreaCommandConfig |
+    CreateAreaCommandConfig;
```

Comment:
This looks suspiciously similar to `messages.proto` where we have:

```
message EditMapMessage {
  oneof message {
    ModifyAreaMessage modifyAreaMessage = 1;
    CreateAreaMessage createAreaMessage = 2;
    DeleteAreaMessage deleteAreaMessage = 3;
  }
}
```

I'm wondering (just wondering) if we could not try to use the interfaces generated by protobuf directly in this package.
I understand it is challenging (because we are in a library and we should therefore generate a library for the protobuf generated package), but this could also save us from doing twice the work (one time in protobuf and one time in map-editor lib). But for this to work, we have to be sure the objects are similar.

What do you think?

Reply:
I would wait for more commands to manifest so we can see if they all follow the same pattern